parachain/primitives/beacon/src/merkleization.rs

```diff
@@ -0,0 +1,44 @@
+use frame_support::log;
+use sp_core::H256;
+use sp_io::hashing::sha2_256;
+
+pub fn is_valid_merkle_branch(
+	leaf: H256,
+	branch: &[H256],
+	depth: u64,
+	index: u64,
+	root: H256,
+) -> bool {
+	if branch.len() as u64 != depth {
```

Comment:
According to https://github.com/ethereum/consensus-specs/blob/dev/ssz/merkle-proofs.md#get_generalized_index_length,  we can calculate depth automatically using `int(log2(index))`.

In rust 1.67 and later, we can use `(branch.len() as u64).checked_ilog2()`.

So we should try this and not bother passing `depth` as a parameter.

Reply:
Simplified a bit to remove depth in this function, but I would suggest not calculate it with logarithm function on the fly.
Btw, index here we provided as const is actually [subtree_index](https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/light-client/sync-protocol.md#get_subtree_index) not the `generalized_index`.